src/commands/auth/sfdxurl/store.ts

```diff
@@ -52,7 +52,12 @@ export default class Store extends SfdxCommand {
   public async run(): Promise<AuthFields> {
     if (await Prompts.shouldExitCommand(this.ux, this.flags.noprompt)) return {};
 
-    const sfdxAuthUrl = await fs.readFile(this.flags.sfdxurlfile, 'utf8');
+    const authFile = this.flags.sfdxurlfile as string;
+
+    const sfdxAuthUrl = authFile.endsWith('.json')
+      ? ((await fs.readJson(authFile)) as Record<string, string>).sfdxAuthUrl
```

Comment:
It would be cool if you could do:

```
sfdx force:org:display --verbose --json > authFile.json
sfdx auth:sfdxurl:store -f authFile.json
```

Could you support that by doing something like?

```
let sfdxAuthUrl =await fs.readFile(authFile, 'utf8');
if (authFile.endsWith('.json')) {
  const authFileJson = parseJson(authFile) as Record<string, string>;
  sfdxAuthUrl = authFileJson.result?.sfdxAuthUrl || authFileJson.sfdxAuthUrl;
}
```

Reply:
Essentially you also need to check the result property.

Reply:
Hmm... perhaps but that introduces more of the same type errors as before
![image](https://user-images.githubusercontent.com/1084688/111011484-9a634800-8367-11eb-9468-ba31fe35b8ce.png)

Reply:
@amphro new commit, how's that?

Not the prettiest but at least encapsulated in its own method.